### PR TITLE
test: base unit test class

### PR DESCRIPTION
**Pull request by armcknight**
While working on failing tests in #2228, I was hitting various crashes due to issues that seemed to be related to statefulness in the test suites. 

This PR creates a new base unit test class that subsumes the old `clearTestState` function we had and automatically runs it on every `setUp`/`tearDown` call–most tests did this, but some would only call it in one or the other.

Other tests had other state management in their `setUp`/`tearDown` that didn't appear in `clearTestState`; those have been added so that they can apply to all tests using this.

Not all tests inherit from this base class yet, only the ones that called `clearTestState`. This is meant to be a step in the direction of getting all our state under control for all tests. I think another step will be moving all of the private `Fixture` classes into this base class so tests are all presented with a consistent mock system, which can be modified as needed depending on the test case. See https://github.com/getsentry/sentry-cocoa/issues/2425

# TODO

Fixed by including a separate class definition for the UI tests. ~- [ ] Figure out how to avoid it being used in UI test targets. Some test classes are present in both UI test and unit test targets.~

#skip-changelog

### CODE REVIEW

File: Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c [hunk 1]
```diff
@@ -394,6 +394,10 @@ sentrycrashfu_removeFile(const char *path, bool mustExist)
 bool
 sentrycrashfu_deleteContentsOfPath(const char *path)
 {
+    if (path == NULL) {
+        return false;
+    }
```
Comment (philipphofmann) [inline Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c:399]: `m`: How is that change related to this PR?

**Review by philipphofmann — APPROVED** [review]
LGTM 💯

File: Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c (hunk 1 above)
Comment (armcknight) [reply Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c:399]: Ah, I forgot to comment on this one. I added a call to `sentrycrash_deleteAllReports` to `clearTestState` (which was done in some of the crash handler unit tests, so I added it here to apply to all tests) but then for test cases that had never written a crash report, this would crash with a NPE. I figured this fix would also just be a nice safeguard to have in production too.

**Review by philipphofmann — APPROVED** [review]
Please merge this soon, and make others changes in a subsequent PR to avoid merge conflicts, as this touches plenty of test classes.

**Comment by philipphofmann** [comment]
@armcknight, what's stopping us from merging this? Can we do additional improvements in forthcoming PRs?

**Comment by armcknight** [comment]
Tests are moving too quick and the merge conflicts are too nasty. I think the better way to do this refactor is going to be to do #2425 first, probably one test class at a time, and then centralize resetting test state afterwards.
